client/services/CodeStoreService.js

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * @fileoverview Service that store code to localStorage
+ */
+tie.factory('CodeStoreService', ['DEFAULT_AUTO_SAVE_SECONDS',
+  function(DEFAULT_AUTO_SAVE_SECONDS) {
+    var codeStoreService = {};
+    var SECONDS_TO_MILLISECONDS = 1000;
+    var getObjFromLocalStorage = function(questionId) {
+      try {
+        return JSON.parse(localStorage.getItem(questionId));
+      } catch (e) {
+        return null;
+      }
+    };
+
+    codeStoreService.saveCode = function(questionId, code, language) {
+      var savedLanguageCodes = getObjFromLocalStorage(
```

Comment:
I think storedCode might be sufficient, here. You seem to be using "saved" and "stored" interchangeably, but it should be consistent within the file.